### PR TITLE
Parameters uniformization in doc

### PR DESCRIPTION
**Pull request by fmonna**
Uniformization of parameter presentation in the documentation.
- put all parameters to @param[in] name format.
- set all function descriptions starting with a verb to infinitive
- set all the parameters descriptions to a start without capitalization

### CODE REVIEW

**Comment by perarnau** [comment]
Nice! I can see a few errors here and there. And clang-format made a mess of course. I'll comment on the errors as I go through the doc.

File: include/aml.h [hunk 1]
```diff
@@ -64,17 +64,17 @@ extern "C" {
 ////////////////////////////////////////////////////////////////////////////////
 
 /**
- * Initializes the library.
- * @param argc: pointer to the main()'s argc argument; contents can get
+ * Initialize the library.
+ * @param[in] argc: pointer to the main()'s argc argument; contents can get
  *        modified.
- * @param argv: pointer to the main()'s argv argument; contents can get
+ * @param[in] argv: pointer to the main()'s argv argument; contents can get
```
Comment (perarnau) [inline include/aml.h:70]:
```suggestion
 * @param[inout] argv: pointer to the main()'s argv argument; contents can get
```
Same for the other.

File: include/aml.h [hunk 2]
```diff
@@ -129,45 +129,44 @@ struct aml_area_ops {
...
-		      void                       *ptr,
-		      size_t                      size);
+	int (*munmap)(const struct aml_area_data *data, void *ptr, size_t size);
 
 	/**
 	 * Print the implementation-specific information available
-	 * @param stream the stream to print to
-	 * @param prefix a prefix string to use on all lines
-	 * @param data non-NULL handle to area internal data.
+	 * @param[out] stream: the stream to print to
```
Comment (perarnau) [inline include/aml.h:162]:
```suggestion
	 * @param[in] stream: the stream to print to
```
That's actually an `in`, even though we "write" into it

File: include/aml.h [hunk 3]
```diff
@@ -186,50 +185,47 @@ struct aml_area {
...
-		void                  *ptr,
-		size_t                 size);
+int aml_area_munmap(const struct aml_area *area, void *ptr, size_t size);
 
 /**
  * Print on the file handle the metadata associated with this area.
- * @param stream the stream to print on
- * @param prefix prefix to use on all lines
- * @param area area to print
+ * @param[out] stream: the stream to print on
```
Comment (perarnau) [inline include/aml.h:212]:
```suggestion
 * @param[in] stream: the stream to print on
```
Same as the function pointer definition. As a rule, there shouldn't be any difference between the generic API and the implementation prototypes, outside of very specific reasons.

File: include/aml.h [hunk 4]
```diff
@@ -360,110 +355,110 @@ struct aml_layout_ops {
...
+	                    const size_t *offsets,
+	                    const size_t *dims,
+	                    const size_t *strides);
 	/**
 	 * Print the implementation-specific information available on a layout,
 	 * content excluded.
-	 * @param stream the stream to print to
-	 * @param prefix a prefix string to use on all lines
-	 * @param data non-NULL handle to layout internal data.
+	 * @param[out] stream: the stream to print to
```
Comment (perarnau) [inline include/aml.h:445]:
```suggestion
	 * @param[in] stream: the stream to print to
```
Idem

File: include/aml.h [hunk 5]
```diff
@@ -616,51 +608,52 @@ int aml_layout_reshape(const struct aml_layout *layout,
...
+                     const size_t *dims,
+                     const size_t *strides);
 
 /**
  * Print on the file handle the metadata associated with this layout.
- * @param stream the stream to print on
- * @param prefix prefix to use on all lines
- * @param layout layout to print
- * @return 0 if successful, an error code otherwise.
+ * @param[out] stream: the stream to print on
```
Comment (perarnau) [inline include/aml.h:631]:
```suggestion
 * @param[in] stream: the stream to print on
```
Idem

File: include/aml.h [hunk 6]
```diff
@@ -616,51 +608,52 @@ int aml_layout_reshape(const struct aml_layout *layout,
...
- * @param[in] src the layout to duplicate
- * @param[out] out a pointer to where to store the new layout
- * @param[in] ptr: If not NULL use this pointer as the new layout raw pointer.
+ * @param[in] src: the layout to duplicate
+ * @param[out] out: a pointer to where to store the new layout
+ * @param[in] ptr: if not NULL use this pointer as the new layout raw pointer.
  * @return -AML_ENOMEM if layout allocation failed.
  * @return -AML_EINVAL if src or dest are NULL.
- * @return AML_SUCCESS if copy succeeded.
+ * @return -AML_SUCCESS if copy succeeded.
```
Comment (perarnau) [inline include/aml.h:648]: Typo ?

File: include/aml.h [hunk 7]
```diff
@@ -753,13 +745,14 @@ struct aml_tiling_ops {
 	/**
 	 * Print the implementation-specific information available on a tiling,
 	 * content excluded.
-	 * @param stream the stream to print to
-	 * @param prefix a prefix string to use on all lines
-	 * @param data non-NULL handle to tiling internal data.
+	 * @param[out] stream: the stream to print to
```
Comment (perarnau) [inline include/aml.h:748]:
```suggestion
	 * @param[in] stream: the stream to print to
```
Idem

File: include/aml.h [hunk 8]
```diff
@@ -848,13 +841,14 @@ struct aml_layout *aml_tiling_index_byiter(const struct aml_tiling *tiling,
 
 /**
  * Print on the file handle the metadata associated with this tiling.
- * @param stream the stream to print on
- * @param prefix prefix to use on all lines
- * @param tiling tiling to print
+ * @param[out] stream: the stream to print on
```
Comment (perarnau) [inline include/aml.h:844]:
```suggestion
 * @param[in] stream: the stream to print on
```
Idem

File: include/aml.h [hunk 9]
```diff
@@ -929,50 +924,54 @@ struct aml_dma_ops {
 	/**
 	 * Initiate a data movement, from a source pointer to a destination
 	 * pointer, and output a request handler for managing the transfer.
-	 * @param dma: dma_implementation internal data.
-	 * @param req[out]: the request handle to manage termination
+	 * @param[in, out] dma: dma_implementation internal data.
+	 * @param[in, out] req: the request handle to manage termination
```
Comment (perarnau) [inline include/aml.h:928]: I think we can keep it just out

File: include/aml.h [hunk 10]
```diff
@@ -929,50 +924,54 @@ struct aml_dma_ops {
...
+	                      void *op_arg);
 
 	/**
 	 * Destroy the request handle. If the data movement is still ongoing,
 	 * then cancel it.
 	 *
-	 * @param dma: dma_implementation internal data.
-	 * @param req: the request handle to manage termination of the movement.
+	 * @param[in] dma: dma_implementation internal data.
+	 * @param[in] req: the request handle to manage termination of the
```
Comment (perarnau) [inline include/aml.h:946]:
```suggestion
	 * @param[inout] req: the request handle to manage termination of the
```
If I remember correctly, this one gets set to NULL on exit.

File: include/aml.h [hunk 11]
```diff
@@ -929,50 +924,54 @@ struct aml_dma_ops {
...
+	                       struct aml_dma_request **req);
 
 	/**
 	 * Wait for termination of a data movement and destroy the request
 	 * handle.
 	 *
-	 * @param dma: dma_implementation internal data.
-	 * @param req: the request handle to manage termination of the movement.
+	 * @param[in] dma: dma_implementation internal data.
+	 * @param[in] req: the request handle to manage termination of the
```
Comment (perarnau) [inline include/aml.h:958]: Same as destroy

File: include/aml.h [hunk 12]
```diff
@@ -929,50 +924,54 @@ struct aml_dma_ops {
...
 	int (*wait_request)(struct aml_dma_data *dma,
-			    struct aml_dma_request **req);
+	                    struct aml_dma_request **req);
 
 	/**
 	 * Print the implementation-specific information available on a dma.
-	 * @param stream the stream to print to
-	 * @param prefix a prefix string to use on all lines
-	 * @param data non-NULL handle to dma internal data.
+	 * @param[out] stream: the stream to print to
```
Comment (perarnau) [inline include/aml.h:967]: The usual

**Review by perarnau — CHANGES_REQUESTED** [review]
Quite nice, I didn't realize we had so many missing metadata in this header. Thanks.
As a general rule, the generic api and the implementation should match. It seems to be the case here for everything. The fprintf one is the same mistake repeated across the file, that's all.

I saw a couple of [out] that could be argued against, but let's leave them as they are for now.

File: include/aml.h (hunk 9 above)
Comment (fmonna) [reply include/aml.h:928]: Ok I'll leave the inout for the dma and switch to out for req

**Review by perarnau — APPROVED** [review]

File: include/aml.h [hunk 13]
```diff
@@ -64,17 +64,17 @@ extern "C" {
 ////////////////////////////////////////////////////////////////////////////////
 
 /**
- * Initializes the library.
- * @param argc: pointer to the main()'s argc argument; contents can get
+ * Initialize the library.
+ * @param[in] argc: pointer to the main()'s argc argument; contents can get
```
Comment (perarnau) [inline include/aml.h:68]:
```suggestion
 * @param[inout] argc: pointer to the main()'s argc argument; contents can get
```

**Review by perarnau — CHANGES_REQUESTED** [review]

**Review by perarnau — APPROVED** [review]

**Pull request merged**